segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -242,4 +245,39 @@ private static void validateContainerIds(Map<Integer, DebugStreamSegmentContaine
                 storage.openWrite(segmentName).thenCompose(segmentHandle -> storage.delete(segmentHandle, TIMEOUT)),
                 ex -> ex instanceof StreamSegmentNotExistsException, null);
     }
+
+    /**
+     * Creates a target segment with the given name and copies the contents of the source segment to the target segment.
+     * @param storage                   A storage instance to create the segment.
+     * @param sourceSegment             The name of the source segment to copy the contents from.
+     * @param targetSegment             The name of the segment to write the contents to.
+     * @param executor                  A thread pool for execution.
+     * @return                          A CompletableFuture that, when completed normally, will indicate the operation
+     * completed. If the operation failed, the future will be failed with the causing exception.
+     */
+    protected static CompletableFuture<Void> copySegment(Storage storage, String sourceSegment, String targetSegment, ExecutorService executor) {
+        AtomicInteger offset = new AtomicInteger();
+        AtomicInteger bytesToRead = new AtomicInteger();
+        return storage.create(targetSegment, TIMEOUT).thenCompose(targetHandle -> {
+            return storage.getStreamSegmentInfo(sourceSegment, TIMEOUT).thenCompose(info -> {
+                return storage.openRead(sourceSegment).thenCompose(sourceHandle -> {
+                    offset.set(0);
```

Comment:
This can  be done in 259 above, when you initialize the variable.

Actually you can move the declaration of both `offset` and `bytesToRead` here.

Reply:
Done.